tests/anmykh/test_homework_6.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+
+from operator import add, mul
+from homework.homework_6.functions import with_required_params
+from anmykh.homework.homework_6 import max_function, min_function, sort_function
+from anmykh.homework.homework_6_reduce_sum import reduce_function, sum_function
+from anmykh.homework.homework_6_unpack import unpack_while_loop, unpack_recursive
+
+@pytest.mark.parametrize(["param1", "param2", "output"], [
+    ["a", "b", ('a', 'b')],
+    [1, 2, (1, 2)],
+    [{"a": 1, "b": 2}, {"b": 1, "a": 2}, ({'a': 1, 'b': 2}, {'a': 2, 'b': 1})]
+])
+def test_with_required_params(param1, param2, output):
+    assert with_required_params(param1, param2) == output
+
+
+@pytest.mark.parametrize(["param", "output"], [
+    [[1], 1],
+    [[-1, 0, 3], 3],
+    [("a"), "a"],
+    [("abc"), "c"]
+])
+def test_max_function(param, output):
+    assert max_function(param) == output
+
+
+@pytest.mark.parametrize(["param"], [
+    [[]],
+    [("")]
+])
+def test_max_function_failure(param):
+    with pytest.raises(ValueError):
+        assert max_function(param)
+
+
+@pytest.mark.parametrize(["param", "output"], [
+    [[1], 1],
+    [[-1, 0, 3], -1],
+    [("a"), "a"],
+    [("abc"), "a"]
+])
+def test_min_function(param, output):
+    assert min_function(param) == output
+
+
+@pytest.mark.parametrize(["param"], [
+    [[]],
+    [("")]
+])
+def test_min_function_failure(param):
+    with pytest.raises(ValueError):
+        assert min_function(param)
```

Comment:
you can use parametrize to test both min and max since they behave similiarly:

```
@pytest.mark.parametrize("func", [
    min_func, 
    max_func
]) 
@pytest.mark.parametrize(["param"], [
    [[]],
    [("")]
])
def test_max_function_failure(func, param):
    with pytest.raises(ValueError):
        assert func(param)

```